internal/provider/data_source_packer_image_iteration_test.go

```diff
@@ -0,0 +1,186 @@
+	createItParams.BucketSlug = bucketSlug
+
+	createItParams.Body = &models.HashicorpCloudPackerCreateIterationRequest{
+		BucketSlug:  bucketSlug,
+		Fingerprint: fingerprint,
+	}
+	_, err := client.Packer.CreateIteration(createItParams, nil)
+	if err == nil {
+		return
+	}
+	if err, ok := err.(*packer_service.CreateIterationDefault); ok {
+		switch err.Code() {
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return
+		}
+	}
+
+	t.Errorf("unexpected CreateIteration error, expected nil or 409. Got %v", err)
+}
+
+func getIterationIDFromFingerPrint(t *testing.T, bucketSlug, fingerprint string) string {
+	t.Helper()
+
+	client := testAccProvider.Meta().(*clients.Client)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	getItParams := packer_service.NewGetIterationParams()
+	getItParams.LocationOrganizationID = loc.OrganizationID
+	getItParams.LocationProjectID = loc.ProjectID
+	getItParams.BucketSlug = bucketSlug
+	getItParams.Fingerprint = &fingerprint
+
+	ok, err := client.Packer.GetIteration(getItParams, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return ok.Payload.Iteration.ID
+}
+
+func upsertBuild(t *testing.T, bucketSlug, fingerprint, iterationID string) {
+	client := testAccProvider.Meta().(*clients.Client)
+
+	createBuildParams := packer_service.NewCreateBuildParams()
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	createBuildParams.LocationOrganizationID = loc.OrganizationID
+	createBuildParams.LocationProjectID = loc.ProjectID
+	createBuildParams.BucketSlug = bucketSlug
+	createBuildParams.BuildIterationID = iterationID
+
+	createBuildParams.Body = &models.HashicorpCloudPackerCreateBuildRequest{
+		Fingerprint: fingerprint,
+		BucketSlug:  bucketSlug,
+		Location:    loc,
+	}
+	createBuildParams.Body.Build = &models.HashicorpCloudPackerBuild{
+		PackerRunUUID: uuid.New().String(),
+		CloudProvider: "aws",
+		ComponentType: "amazon-ebs.example",
+		IterationID:   iterationID,
+		Status:        models.HashicorpCloudPackerBuildStatusDONE,
+		Images: []*models.HashicorpCloudPackerImage{
+			{
+				ImageID: "ami-42",
+			},
+			{
+				ImageID: "ami-43",
+			},
+		},
+	}
+	_, err := client.Packer.CreateBuild(createBuildParams, nil)
+	if err == nil {
+		return
+	}
+	if err, ok := err.(*packer_service.CreateBuildDefault); ok {
+		switch err.Code() {
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return
+		}
+	}
+
+	t.Errorf("unexpected CreateBuild error, expected nil or 409. Got %v", err)
+}
+
+func TestAcc_dataSourcePacker(t *testing.T) {
+	resourceName := "data.hcp_packer_image_iteration.alpine"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t, false) },
+		ProviderFactories: providerFactories,
+
```

Comment:
Since this test upserts some test data that TF doesn't manage directly, we'll need an extra step to clean them up. You can add `CheckDestroy: testAccPackerCheckDestroy` here, then implement the CheckDestroy function to remove all the upserted data and verify the data source is deleted ([peering example](https://github.com/hashicorp/terraform-provider-hcp/blob/main/internal/provider/resource_aws_network_peering_test.go#L80))

Reply:
@azr this has been implemented in #180 which @bcmdarroch has reviewed and approved. The change contain a few additional error checks, and a potential bug fix when requesting a channel that has not been assigned a complete iteration. 

I confirmed the panic using the provider on an exiting channel with no assigned iteration, and confirmed that the fix errors as expected. Merging the changes as they are for now so that we can get this in. But please follow up on the changes.

Reply:
Thanks Wilken !